apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -22,17 +22,30 @@ defmodule GameBackend.CurseOfMirra.Quests do
   end
 
   @doc """
-  Get all %Quest{} by the type field
+  Get all %Quest{} by the type field that doesn't have a valid daily quest
+
+  a valid daily quest means:
+  - the inserted at is inside the current day period
+  - It hasn't been completed
 
   ## Examples
 
-      iex>get_quest_by_config_id("daily")
+      iex>get_user_missing_quests_by_type(user_id, "daily")
       [%Quest{type: "daily"}]
   """
-  def get_quests_by_type(type) do
+  def get_user_missing_quests_by_type(user_id, type) do
+    naive_today = NaiveDateTime.utc_now()
+    start_of_date = NaiveDateTime.beginning_of_day(naive_today)
+    end_of_date = NaiveDateTime.end_of_day(naive_today)
+
     q =
-      from(qd in Quest,
-        where: qd.type == ^type
+      from(q in Quest,
+        left_join: dq in DailyQuest,
+        on: q.id == dq.quest_id and dq.user_id == ^user_id,
+        where:
+          (is_nil(dq) or dq.inserted_at < ^start_of_date or dq.inserted_at > ^end_of_date or not is_nil(dq.completed_at)) and
```

Comment:
Documentation says the daily quests shouldn't be completed but here you're checking that the `:completed_at` field is not nil, that means that the quest has been completed.

Reply:
that's right, the thing of this is that we're negating the valid quests:
`that doesn't have a valid daily quest`
so we're looking for quests from other days or completed quests